main.go

```diff
@@ -315,3 +318,61 @@ func loggerFromFlags(cmd *cli.Command) *slog.Logger {
 	}
 	return slog.New(h)
 }
+
+// metrics configuration
+func newMetrics() *metrics.Metrics {
+	return &metrics.Metrics{
+		TMIMsgsCount: metrics.NewPromCounter(
+			promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
With `api` taking the collectors to serve, there's no longer any actual benefit to using promauto. These can change to prometheus.NewCounter instead.